### PR TITLE
FIXES #81: Add tests for checking color of spinner in stdout

### LINKED ISSUE #81: Add tests for checking color of spinner in stdout
## Description

Currently tests run on cleaned `stdout` with no ansi escapings. Actual color in output is not tested. This issue aims to track and fix it.

See https://github.com/manrajgrover/halo/pull/78#pullrequestreview-155713163 for more information.

## People to notify

@manrajgrover

### PR DESCRIPTION
**Pull request by eoinnoble**
## Description of new feature, or changes
`_get_test_output` has been refactored to take a `no_ansi` arg,
which defaults to `True`. We then use this in the new test to keep
the ANSI escape characters and compare them to those supplied by
colorama.

It feels a little like we're testing another library with this
test but I can understand why you feel like you need to have it
in there.

I am only able to run the py36 tests at present, and they **do not
currently pass for me** (see [my other PR](https://github.com/manrajgrover/halo/pull/99) for a fix for that test). To
keep this PR clean I have left them like that. The test that I have
written obviously passes, though.

## Checklist

- [x] Your branch is up-to-date with the base branch
- [x] You've included at least one test if this is a new feature
- [x] All tests are passing

## Related Issues and Discussions
#81 

## People to notify
@manrajgrover

### CODE REVIEW

File: tests/test_halo.py [hunk 1]
```diff
@@ -441,6 +442,25 @@ def test_animation_setter(self):
...
+        }
+
+        for k, v in colors.items():
+            self._stream = io.open(self._stream_file, 'w+')  # reset stream
+            spinner = Halo(color=k, stream=self._stream)
+            spinner.start()
+            spinner.stop()
+
+            output = self._get_test_output(no_ansi=False)
+            assert v in output[1]
```
Comment (manrajgrover) [inline tests/test_halo.py]: @eoinnoble We should do this here:

```suggestion
            self.assertEquals(color_ascii, output[1])
```

File: tests/test_halo.py [hunk 2]
```diff
@@ -441,6 +442,25 @@ def test_animation_setter(self):
...
+        """Test ANSI escape characters are present
+        """
+        colors = {
+            'cyan': Fore.CYAN,
+            'magenta': Fore.MAGENTA,
+            'yellow': Fore.YELLOW,
+            'blue': Fore.BLUE
+        }
+
+        for k, v in colors.items():
```
Comment (manrajgrover) [inline tests/test_halo.py]: @eoinnoble Lets make this:

```suggestion
        for color_name, color_ascii in colors.items():
```

File: tests/test_halo.py [hunk 3]
```diff
@@ -441,6 +442,25 @@ def test_animation_setter(self):
...
+            self._stream = io.open(self._stream_file, 'w+')  # reset stream
+            spinner = Halo(color=k, stream=self._stream)
```
Comment (manrajgrover) [inline tests/test_halo.py]: @eoinnoble This will become

```suggestion
            spinner = Halo(color=color_name, stream=self._stream)
```

**Comment by manrajgrover** [comment]
@eoinnoble Looks like test cases are failing. Could you please look into it?

**Comment by eoinnoble** [comment]
It was the `assert` statement, should be `self.assertEquals(color_ascii in output[1], True)` if you don't want to use a bare one. I'm not sure if git has picked up my change, can you restart the automated checks from your end?

**Comment by eoinnoble** [comment]
It was the `assert` statement, should be `self.assertEquals(color_ascii in output[1], True)` if you don't want to use a bare one. I'm not sure if git has picked up my change, can you restart the automated checks from your end?

**Comment by eoinnoble** [comment]
It was the `assert` statement, should be `self.assertEquals(color_ascii in output[1], True)` if you don't want to use a bare one. I'm not sure if git has picked up my change, can you restart the automated checks from your end?

**Comment by eoinnoble** [comment]
It was the `assert` statement, should be `self.assertEquals(color_ascii in output[1], True)` if you don't want to use a bare one. I'm not sure if git has picked up my change, can you restart the automated checks from your end?

**Comment by manrajgrover** [comment]
@eoinnoble Thanks for fixing. Currently, Github is facing a lot of issues. Once they are fixed, I'll rerun Travis.

**Comment by manrajgrover** [comment]
Hi @eoinnoble, apologies for the delay! Thanks for working on it! :100:

We should also test halo notebook. I'll keep the issue open if you're up for it.

**Comment by eoinnoble** [comment]
Sure, I can take a stab at that over the coming days.

**Pull request merged**